src/LercLib/RLE.cpp

```diff
@@ -315,12 +315,20 @@ bool RLE::decompress(const Byte* arrRLE, size_t nBytesRemaining, Byte* arr, size
 
     if (cnt > 0)
     {
-      while (i--) arr[arrIdx++] = *srcPtr++;
+      while (i)

```

Comment:
Would line 310 also be part of this problem?  `size_t` is always expected to be unsigned, but in some cases you've got `-cnt` being set.

Reply:
no problem with line 310. It takes the absolute value of cnt, so the result is positive or zero

Reply:
True. But how about we change line 310 to "int i = ..."? Then your uint overflow warning should go away. Perhaps might trigger the need for a (size_t) cast in line 311? If not that would be the smaller code change.

Reply:
Done